src/spec.ts

```diff
@@ -59,6 +59,14 @@ export interface BaseSpec {
   transform?: Transform[];
 }
 
+export type DataRequired = {
```

Comment:
Make this `DataMixins` and re-use it elsewhere with `Partial<DataMixins>` to make sure that the documentation is always consistent if we update them in the future?

Reply:
That requires some non-trivial changes in the schema compiler. I will do that after the deadline.